packages/xrpl/src/models/transactions/transaction.ts

```diff
@@ -97,9 +97,11 @@ export type Transaction =
 /**
  * @category Transaction Models
  */
-export interface TransactionAndMetadata {
-  transaction: Transaction
-  metadata: TransactionMetadata
+export interface TransactionAndMetadata<
+  T extends BaseTransaction = Transaction,
+> {
+  transaction: T
```

Comment:
nit: The tricky part of modifying this type is that `ledger` command and streams will not return these nice new properties.  Might be worth adding a comment about that.